packages/storage/amplify_storage_s3/example/lib/main.dart

```diff
@@ -105,19 +108,271 @@ class HomeScreen extends StatefulWidget {
 }
 
 class _HomeScreenState extends State<HomeScreen> {
+  var list = [];
+  var imageUrl = '';
+
+  @override
+  void initState() {
+    super.initState();
+    _checkAuthStatus();
+    _listAllPublicFiles();
+  }
+
+  // sign out of the app
+  Future<void> _signOut() async {
+    try {
+      await Amplify.Auth.signOut();
+      _logger.debug('Signed out');
+    } on AuthException catch (e) {
+      _logger.debug('Could not sign out - ${e.message}');
+    }
+  }
+
+  // check if the user is signed in
+  Future<void> _checkAuthStatus() async {
+    try {
+      final session = await Amplify.Auth.fetchAuthSession();
+      _logger.debug('Signed in: ${session.isSignedIn}');
+    } on AuthException catch (e) {
+      _logger.debug('Could not check auth status - ${e.message}');
+    }
+  }
+
+  // upload a file to the S3 bucket
+  Future<void> _uploadFile() async {
+    final result = await FilePicker.platform.pickFiles(
+      type: FileType.custom,
+      allowedExtensions: ['jpg', 'png'],
+      withReadStream: true,
+      withData: false,
+    );
+
+    if (result == null) {
+      _logger.debug('No file selected');
+      return;
+    }
+
+    final platformFile = result.files.single;
+
+    try {
+      final result = await Amplify.Storage.uploadFile(
```

Comment:
```suggestion
      await Amplify.Storage.uploadFile(
```
nit: a couple of these functions are being saved to `result` but are never used.